src/wallet/wallet.h

```diff
@@ -919,7 +919,7 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     void BlockDisconnected(const std::shared_ptr<const CBlock>& pblock) override;
     bool AddToWalletIfInvolvingMe(const CTransactionRef& tx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate);
     int64_t RescanFromTime(int64_t startTime, bool update);
-    CBlockIndex* ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate = false);
+    CBlockIndex* ScanForWalletTransactions(CBlockIndex* pindexStart, CBlockIndex* pindexStop = nullptr, bool fUpdate = false);
```

Comment:
It looks like you've updated all the calls to `ScanForWalletTransactions()` to explicitly set `pindexStop` (except in `wallet_tests.cpp`). Why not update `wallet_tests.cpp` and remove the default argument. I think in general it's best to avoid default arguments if possible.